rest/handler.go

```diff
@@ -209,17 +213,24 @@ func (h *handler) invoke(method handlerMethod, accessPermissions []Permission, r
 		if dbContext, err = h.server.GetDatabase(dbname); err != nil {
 			base.Infof(base.KeyHTTP, "Error trying to get db %s: %v", base.MD(dbname), err)
 
-			if shouldCheckAdminAuth {
-				if httpError, ok := err.(*base.HTTPError); ok && httpError.Status == http.StatusNotFound {
+			if httpError, ok := err.(*base.HTTPError); ok && httpError.Status == http.StatusNotFound {
+				if shouldCheckAdminAuth {
 					authorized, err := h.checkAdminAuthenticationOnly()
 					if err != nil {
 						return err
 					}
 
-					if authorized {
-						return base.HTTPErrorf(http.StatusForbidden, "")
+					if !authorized || h.providedAuthCredentials() {
+						return ErrInvalidLogin
+					}
+				} else {
+					if h.privs == regularPrivs || h.privs == publicPrivs {
+						if !h.providedAuthCredentials() {
+
+							return ErrLoginRequired
+						}
+						return ErrInvalidLogin
```

Comment:
I'm not sure why this change is different than the one that went into `master` ?

https://github.com/couchbase/sync_gateway/pull/6183/files#diff-9899f6c02129d809f5816c71f72902fd77314b35b000f7484d4ba6ea2e5ca1ef